File: Painless/Distinct.cpp
#include <algorithm>

int solution(vector<int> &A) {

    int n=A.size();
    if(n==0) return 0;
    
    int count=1;
    sort(A.begin(),A.end()); //nLog(n)
    for(int i =0;i<n-1;i++)
    {
        if(A[i]!=A[i+1]){count++;}    
    }
    return count;
}

File: Painless/Distinct.py
def solution(A):
    # write your code in Python 3.6
    A.sort()
    if len(A) == 0 :
        return 0
    last = A[0]
    counter=1
    for element in A:
        if element != last:
            counter+=1
            last =element
    return counter
